src/test/groovy/graphql/execution/ResultPathTest.groovy

```diff
@@ -212,6 +212,13 @@ class ResultPathTest extends Specification {
         path.toList() == ["a", "b"]
     }
 
+    def "pass any other object than string or int"(){
+        when:
+        ResultPath.fromList(["a", "b", true])
```

Comment:
the graphql-java code itself would never use anything but a String or Integer - but I am happy to have this be more robust